app/src/Controller/ArticlesController.php

```diff
@@ -0,0 +1,131 @@
+<?php
+declare(strict_types=1);
+
+namespace App\Controller;
+
+use Cake\Datasource\EntityInterface;
+
+/**
+ * Articles Controller
+ *
+ * @property \App\Model\Table\ArticlesTable $Articles
+ */
+class ArticlesController extends AppController
+{
+    /**
+     * Initialize method.
+     *
+     * @return void
+     */
+    public function initialize(): void
+    {
+        parent::initialize();
+
+        $this->loadComponent('Paginator');
+        $this->loadComponent('Flash'); // Include the FlashComponent
+    }
+
+    /**
+     * Index method.
+     *
+     * @return \Cake\Http\Response|null|void Renders view
+     */
+    public function index()
+    {
+        $articles = $this->Paginator->paginate($this->Articles->find());
+        $this->set(compact('articles'));
+    }
+
+    /**
+     * View method.
+     *
+     * @param string|null $slug Article slug
+     * @return \Cake\Http\Response|null|void Renders view
+     */
+    public function view($slug = null)
+    {
+        $article = $this->Articles->find()
+            ->where(['slug' => $slug])
+            ->firstOrFail();
+        $this->set(compact('article'));
+    }
+
+    /**
+     * Add method.
+     *
+     * @return \Cake\Http\Response|null|void Redirects on successful add
+     */
+    public function add()
+    {
+        $article = $this->Articles->newEmptyEntity();
+        if ($this->request->is('post')) {
+            $article = $this->Articles->patchEntity($article, $this->request->getData());
+
+            // Hardcoding the user_id is temporary and will be removed later
+            // when we build authentication out.
```

Comment:
The comment itself requires to be fixed later. Adding TODO is easily noticeable and will make other members aware there's incomplete work that needs to be done here.